server/pbench/bin/pbench-sync-satellite

```diff
@@ -123,66 +150,100 @@ for host in $hosts ;do
     fi
 
     # move the unpacked files from tmp directory to archive
-    mv $unpack/$host/{.,}* .
-    rc=$?
-    if [[ $rc != 0 ]] ;then
-        echo "FAILED:    mv $unpack/$host/* ."
-        let failures=failures+1
-    fi
+    mv $unpack/$host/* $unpack/$host/.prefix* .
 
     # make the state dirs: TODO, TO-INDEX, TO-COPY-SOS etc.
     mk_dirs $prefix::$host
 
     # check md5s and move md5s to its appropriate state directories according to pass or fail
     md5_list=$(find . -path '*.tar.xz.md5' -printf '%P\n')
     md5sum -c $md5_list > $logdir/$host/md5-checks.log
+    processed=$(wc -l < $logdir/$host/md5-checks.log)
+    nprocessed=$((nprocessed + processed))
     grep 'OK' $logdir/$host/md5-checks.log > $logdir/$host/ok-checks.log
     if [ -s $logdir/$host/ok-checks.log ] ;then
         md5_pass=$(cat $logdir/$host/ok-checks.log | sed -n 's/: OK//p')
         for x in $md5_pass; do
-            ln -s $x SATELLITE-MD5-PASSED/$x
+            ln -sf $PWD/$x SATELLITE-MD5-PASSED/$x
+            status=$?
+            if [[ $status != 0 ]] ;then
+                nerrs=$nerrs+1
+                echo "Failed to create the symlink for md5 check passed tarballs: ln -sf \
+                $PWD/$x SATELLITE-MD5-PASSED/$x" >&4 | tee -a $mail_content
```

Comment:
I don't think this works: you have to say

      echo "...." | tee -a $mail_content >&4

otherwise the output will not get into $mail_content. That should be changed everywhere in this file (and maybe also in PR#677: please check). I always try small experiments on the command line to make sure that these constructs work the way I think they do.